aws/table_aws_waf_rule.go

```diff
@@ -0,0 +1,226 @@
+				Description: "The ID of the Rule.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "metric_name",
+				Description: "The name of the metric for the Rule.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getAwsWAFRule,
+			},
+			{
+				Name:        "predicates",
+				Description: "The Predicates object contains one Predicate element for each ByteMatchSet,IPSet, or SqlInjectionMatchSet object that you want to include in a Rule.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsWAFRule,
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags assigned to the Rule.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsWAFRuleTag,
+				Transform:   transform.FromField("TagInfoForResource.TagList"),
+			},
+
+			// Standard columns for all tables
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsWAFRuleTag,
+				Transform:   transform.FromField("TagInfoForResource.TagList").Transform(wafRuleTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getAwsWAFRuleAkas,
+				Transform:   transform.FromValue(),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsWAFRules(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	plugin.Logger(ctx).Trace("listAwsWAFRules")
+
+	// Create session
+	svc, err := WAFService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// List call
+	params := &waf.ListRulesInput{}
+	pagesLeft := true
+	for pagesLeft {
+		response, err := svc.ListRules(params)
+		if err != nil {
+			return nil, err
+		}
+		for _, rule := range response.Rules {
+			d.StreamListItem(ctx, rule)
+		}
+		if response.NextMarker != nil {
+			pagesLeft = true
+			params.NextMarker = response.NextMarker
+		} else {
+			pagesLeft = false
+		}
+
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsWAFRule(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("getAwsWAFRule")
+
+	// Create Session
+	svc, err := WAFService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	var id string
+	if h.Item != nil {
+		id = ruleData(h.Item)
+	} else {
+		id = d.KeyColumnQuals["rule_id"].GetStringValue()
+	}
+
+	// Build the params
+	param := &waf.GetRuleInput{
+		RuleId: aws.String(id),
+	}
+
+	// Get call
+	data, err := svc.GetRule(param)
+	if err != nil {
+		return nil, err
+	}
+
+	return data.Rule, nil
+}
+
+func getAwsWAFRuleTag(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment:
```suggestion
func getAwsWAFRuleTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

I think `Tags` should be plural since we could be getting more than one.